subscription-fee/src/subtract_payments.rs

```diff
@@ -0,0 +1,149 @@
+use core::hint::unreachable_unchecked;
+
+use auto_farm::common::{address_to_id_mapper::AddressId, unique_payments::UniquePayments};
+
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+#[must_use]
+#[derive(Debug, PartialEq, Eq, Clone, TopEncode, TopDecode, TypeAbi)]
+pub enum MyVeryOwnScResult<
+    T: NestedEncode + NestedDecode + TypeAbi,
+    E: NestedEncode + NestedDecode + TypeAbi,
+> {
+    Ok(T),
+    Err(E),
+}
+
+impl<T: NestedEncode + NestedDecode + TypeAbi, E: NestedEncode + NestedDecode + TypeAbi>
+    From<Result<T, E>> for MyVeryOwnScResult<T, E>
+{
+    fn from(value: Result<T, E>) -> Self {
+        match value {
+            Result::Ok(t) => MyVeryOwnScResult::Ok(t),
+            Result::Err(e) => MyVeryOwnScResult::Err(e),
+        }
+    }
+}
+
+impl<T: NestedEncode + NestedDecode + TypeAbi, E: NestedEncode + NestedDecode + TypeAbi>
+    MyVeryOwnScResult<T, E>
+{
+    pub fn is_err(&self) -> bool {
+        matches!(*self, MyVeryOwnScResult::Err(_))
+    }
+
+    /// # Safety
+    ///
+    /// Calling this method on an [`Err`] is *[undefined behavior]*.
+    pub unsafe fn unwrap_unchecked(self) -> T {
+        match self {
+            MyVeryOwnScResult::Ok(t) => t,
+            MyVeryOwnScResult::Err(_) => unreachable_unchecked(),
+        }
+    }
+}
```

Comment:
Would be better to use addresses instead of IDs for arguments on endpoints. For external integrations is easier to work directly with address instead of ID